### PR TITLE
Actually creating unit tests

### PR DESCRIPTION
**Pull request by kenny2892**

### CODE REVIEW

File: Trivia Maze/src/application/Main.java [hunk 1]
```diff
@@ -86,7 +86,6 @@ else if (stage == null)
 		gameMaze = new Maze(ROWS, COLS);
 		player = new Player();
 		database = new Database(type);
-		dataType = type;
```
Comment (kenny2892) [inline Trivia Maze/src/application/Main.java]: Why remove the dataType assignment? I use that when picking what bg music should be playing.

File: Trivia Maze/src/application/Main.java [hunk 2]
```diff
@@ -433,10 +433,8 @@ public static void setCheatAnswer(boolean cheat)
 	
 	public static void setHorizontalArchway(int y, int x, ArchwayStatus status)
 	{
-		if (stage == null || gameMaze == null)
-			return;
 		
-		else if(status == null)
+		if(status == null)
```
Comment (kenny2892) [inline Trivia Maze/src/application/Main.java:438]: Should also check if horizontalArchways is null.

File: Trivia Maze/src/application/Main.java [hunk 3]
```diff
@@ -450,10 +448,7 @@ else if(x >= COLS || x < 0)
 	
 	public static void setVerticalArchway(int y, int x, ArchwayStatus status)
 	{
-		if (stage == null || gameMaze == null)
-			return;
-		
-		else if(status == null)
+		if(status == null)
```
Comment (kenny2892) [inline Trivia Maze/src/application/Main.java:454]: Should also check if verticalArchways is null.

File: Trivia Maze/src/application/Maze.java [hunk 4]
```diff
@@ -49,6 +45,11 @@ public Maze(int mazeRows, int mazeColumns)
 
 	public boolean updateMazeRooms(int x, int y, boolean isLocked)
 	{
+		if (x < 0 || x >= 5)
```
Comment (kenny2892) [inline Trivia Maze/src/application/Maze.java]: Should change from using 5 to using Main.ROWS and Main.COLUMNS

File: Trivia Maze/src/application/Maze.java [hunk 5]
```diff
@@ -186,20 +187,22 @@ public Room getRoom(int x, int y)
...
 	{
+		if(newDirection == null)
+			throw new IllegalArgumentException("Direction was null");
 		this.currentDirection = newDirection;
 	}
 
 	public void setExitX(int exitX)
 	{
-		if (exitX < 0 || exitX >= this.mazeColumns)
+		if (exitX < 0 || exitX >= 5)
```
Comment (kenny2892) [inline Trivia Maze/src/application/Maze.java]: Should change from using 5 to using Main.COLUMNS

File: Trivia Maze/src/application/Maze.java [hunk 6]
```diff
@@ -186,20 +187,22 @@ public Room getRoom(int x, int y)
...
 			throw new IllegalArgumentException("Passed X value is out of the bounds of the maze.");
 		
 		this.exitX = exitX;
 	}
 
 	public void setExitY(int exitY)
 	{
-		if (exitY < 0 || exitY >= this.mazeRows)
+		if (exitY < 0 || exitY >= 5)
```
Comment (kenny2892) [inline Trivia Maze/src/application/Maze.java]: Should change from using 5 to using Main.ROWS

File: Trivia Maze/src/application/Main.java (hunk 2 above)
Comment (HMRich) [reply Trivia Maze/src/application/Main.java:438]: Those are two separate methods.

File: Trivia Maze/src/application/Main.java (hunk 3 above)
Comment (HMRich) [reply Trivia Maze/src/application/Main.java:454]: Those are two separate methods.

File: Trivia Maze/src/application/Maze.java (hunk 5 above)
Comment (HMRich) [reply Trivia Maze/src/application/Maze.java]: Sure this program is not designed to scale.

File: Trivia Maze/src/application/Maze.java (hunk 6 above)
Comment (HMRich) [reply Trivia Maze/src/application/Maze.java]: Sure this program is not designed to scale.

**Pull request merged**